agray0232/landmarks
# Dynamic Login/Logout Button

**Description**
Applied dynamic login and logout buttons in all of the page's navigation bars. When a user is not logged in, the button is "Login" which will take them to a login page. When a user is not logged in, the button is "Logout" which will log them out.

Additionally, the "Get Started" button was made dynamic so that if a user is not logged in, it will open the login prompt, but if a user is logged in, it will redirect to the Trips page.

**Changes**
Oh so many changes...
* I restructured how the Firebase management was done. Put everything into an authentication.js file
* Tried to standardize how each page initialized and updated
* Added event listener management
* Added rendering to the below buttons
* Created dynamic Login/logout buttons
* Create dynamic Get Started button
* Broke some functions out into smaller functions
* Commented everything

**Test Procedure**
If you are reviewing this, please checkout this branch and make sure everything works well
